Apps/Common/src/Delegates/WalletIssuerDelegate.cs

```diff
@@ -119,10 +123,83 @@ public async Task<RequestResult<ConnectionResponse>> CreateConnectionAsync(Guid
+
+                    RequestResult<CredentialDefinitionIdResponse> credentialDefinitionIdResponse = await this.GetCredentialDefinitionIdsAsync(schemaId).ConfigureAwait(true);
+                    if (credentialDefinitionIdResponse.ResultStatus == ResultType.Success)
+                    {
+                        string credentialDefinitionId = credentialDefinitionIdResponse.ResourcePayload!.CredentialDefinitionIds.First();
+                        CredentialProposal credentialProposal = new ();
+                        foreach (PropertyInfo property in payload.GetType().GetProperties())
+                        {
+                            if (property != null)
+                            {
+                                credentialProposal.Attributes.Add(new CredentialAttribute
+                                {
+                                    Name = property.Name,
+                                    Value = property.GetValue(payload)?.ToString(),
+                                });
+                            }
+                        }
+
+                        CredentialOfferRequest credentialOffer = new CredentialOfferRequest
+                        {
+                            ConnectionId = connection.AgentId,
+                            IssuerDid = did,
+                            SchemaId = schemaId,
+                            SchemaIssuerDid = did,
+                            SchemaName = this.WalletIssuerConfig.SchemaName,
+                            SchemaVersion = this.WalletIssuerConfig.SchemaVersion,
+                            CredentialDefinitionId = credentialDefinitionId,
+                            Comment = comment,
+                            CredentialProposal = credentialProposal,
+                        };
+
+                        RequestResult<CredentialResponse> credentialResponse = await this.IssueCredentialAsync(credentialOffer).ConfigureAwait(true);
+                        if (credentialResponse.ResultStatus == ResultType.Success)
+                        {
+                            retVal.ResourcePayload = credentialResponse.ResourcePayload;
+                            retVal.TotalResultCount = 1;
+                            retVal.ResultStatus = ResultType.Success;
+                        }
+                        else
+                        {
+                            retVal.ResultError = credentialResponse.ResultError;
+                        }
+                    }
+                    else
+                    {
+                        retVal.ResultError = credentialDefinitionIdResponse.ResultError;
+                    }
+                }
+                else
+                {
+                    retVal.ResultError = schemaIdResponse.ResultError;
+                }
+            }
+            else
+            {
+                retVal.ResultError = issuerDidResponse.ResultError;
+            }
+
+            return retVal;
```

Comment:
This would have fewer indentations and a smaller complexity rating if the checks were inverted and returned early.